src/mam/mod_mam_meta.erl

```diff
@@ -168,6 +168,11 @@ parse_backend_opts(elasticsearch, Type, Opts, Deps0) ->
         _ -> Deps
     end.
 
+% muc backend requires both pm and muc user DB to populate sender_id column
+-spec user_db_types(pm | muc) -> [pm | muc].
```

Comment:
Correct me if I am wrong but this spec should be:
```erlang
-spec user_db_types(pm | muc) -> [pm] | [pm, muc].
```
??

Reply:
because current spec allows for the function to return `[muc]`?

Reply:
`[pm, muc]` is illegal.

Reply:
oh sorry, missed that